gradle-incremental-configuration-cache/src/main/java/com/palantir/gradle/configcache/incremental/IncrementalConfigurationCachePlugin.java

```diff
@@ -31,29 +31,25 @@ public class IncrementalConfigurationCachePlugin implements Plugin<Project> {
 
     @Override
     public final void apply(Project project) {
-        Path allowlistPath = project.getRootProject().getProjectDir().toPath().resolve(ALLOW_LIST_FILE);
-        if (!Files.exists(allowlistPath)) {
-            log.warn(
-                    "Configuration cache allowed tasks file not found at: {}. " +
-                            "All tasks will be marked as incompatible with configuration cache.",
-                    allowlistPath.toAbsolutePath()
-            );
-            return;
+        if (!project.getRootProject().equals(project)) {
+            throw new RuntimeException("Must be applied only to root project");
         }
 
-        project.afterEvaluate(_proj -> {
-            AllowListFile allowList = new AllowListFile(allowlistPath);
-            Set<String> enabledTasks = allowList.loadAllowedTasks();
-
-            project.getTasks().configureEach(task -> {
-                if (!enabledTasks.contains(task.getPath())) {
-                    System.out.println("CC disabled for: " + task.getName());
-                    task.notCompatibleWithConfigurationCache(
-                            String.format(
-                                    "Configuration cache is not enabled for this task, as it was not included in %s",
-                                    ALLOW_LIST_FILE));
-                }
-            });
-        });
+        Path allowListPath = project.getRootProject().getProjectDir().toPath().resolve(ALLOW_LIST_FILE);
+        if (!Files.exists(allowListPath)) {
+            throw new RuntimeException(
+                    String.format("Configuration cache allowed tasks file not found at %s", allowListPath));
```

Comment:
Since we're on java 17 now, you can actually do:

```java
"Configuration cache allowed tasks file not found at %s".formatted(allowListPath)
```

If you'd prefer